spacy/cli/debug_data.py

```diff
@@ -670,6 +671,38 @@ def debug_data(
                 f"Found {gold_train_data['n_cycles']} projectivized train sentence(s) with cycles"
             )
 
+    if "trainable_lemmatizer" in factory_names:
+        msg.divider("Trainable Lemmatizer")
+        # NOTE: label_list is the string version of the trees
+        label_list: Set[str] = gold_train_data["lemmatizer_trees"]
+
+        # NOTE: label_list_full is the set of strigified trees from
+        # iterating through all trees after all lemmas have been added
+        label_list_full: Set[str] = gold_train_data["lemmatizer_trees_full"]
+
+        # NOTE: model_labels is the tree IDs from iterating through the tree_id
+        # values in the component.
+        model_labels = _get_labels_from_trainable_lemmatizer(nlp)
+
+        msg.info(
+            f"{len(label_list)} label(s) in train data (created from trees.add and trees.tree_to_str)"
+        )
+        msg.info(
+            f"{len(label_list_full)} label(s) in train data (from iterating through tree_id)"
+        )
+        msg.info(
+            f"{len(model_labels)} label(s) in component (from iterating through tree_id on component.trees"
+        )
+
+        labels = set(label_list)
+        missing_labels = model_labels - labels
+
+        if missing_labels:
+            msg.warn(
+                f"{len(missing_labels)} label(s) in model (component.tree)"
+                " and not present in the train data (trees.add and trees.tree_to_str)."
+            )
+
```

Comment:
Current output run on example project in PR description:

```
Output from run on 

```
ℹ 565 label(s) in train data (created from trees.add and
trees.tree_to_str)
ℹ 1120 label(s) in train data (from iterating through tree_id)
ℹ 510 label(s) in component (from iterating through tree_id on
component.trees
⚠ 275 label(s) in model (component.tree) and not present in the train
data (trees.add and trees.tree_to_str).
```

Reply:
These messages/numbers aren't going to meaningful to users, who don't know anything about the internals (and shouldn't need to know anything about the internals).

In general there are a couple different kinds of missing labels to consider:

* you have labels in the dev data that aren't in the train data
* you're updating a sourced model with existing labels and your current training data only includes a subset of the labels, so you'll run into catastrophic forgetting

I think the last line here is looking at yet another kind of missing label, which is edit trees that are omitted for a newly initialized component due to a threshold in the component config, which might not matter at all for coverage of the dev data. It's not that they're not present in the train data (there's nowhere else for these edit trees to have come from), but that they're not included in the initialized component even though they're in the train data. Does that make sense?

Numbers related to the edit tree counts probably aren't very informative for users, but what would be interesting is the number of dev lemmas that can be predicted at all from the current edit trees in the component, which is the first kind of missing label above, similar to a tagger that has `ADJ` in the dev data but not in the train data.

And looking at `debug data` again, I think it's not handling the initialization correctly for sourced components, so anything related to the second kind of missing label isn't working at all right now.

Reply:
> These messages/numbers aren't going to meaningful to users, who don't know anything about the internals (and shouldn't need to know anything about the internals).

I agree these aren't helpful, I was putting these and other explanatory comments in here as a draft for myself to better understand how these numbers lined up, and would remove them before review.

> It's not that they're not present in the train data (there's nowhere else for these edit trees to have come from), but that they're not included in the initialized component even though they're in the train data. Does that make sense?

I think so. If we've got component labels `CL` and train data labels `TL`, this questions is about `TL - CL`, which might be not empty because of some component config threshold. The set of `CL - TL` with a sourced component results in catastrophic forgetting.

> Numbers related to the edit tree counts probably aren't very informative for users, but what would be interesting is the number of dev lemmas that can be predicted at all from the current edit trees in the component, which is the first kind of missing label above, similar to a tagger that has ADJ in the dev data but not in the train data.
>
> And looking at debug data again, I think it's not handling the initialization correctly for sourced components, so anything related to the second kind of missing label isn't working at all right now.

It seems like we should prioritize the first kind then, and ignore the second kind for now?

Reply:
> which is the first kind of missing label above, similar to a tagger that has `ADJ` in the dev data but not in the train data.

We do this with `deps`, but not for `tags`, right? https://github.com/explosion/spaCy/pull/11419/files#diff-237bbf8610a0a5f63114db5e097db1325b2f4c11c1a8f79f9e20b41e6ec0e8cbR631-R645